### PR TITLE
Refactor QUIC new connection handler function

### PR DESCRIPTION
**Pull request by pgarg66**
#### Problem
`setup_connection` function has gotten convoluted over time, and could use a refactor.

#### Summary of Changes
Refactored the function into smaller blocks and also improve the code reuse.

Fixes #

### CODE REVIEW

**Comment by KirillLykov** [comment]
Generally, it is a good idea to simplify original code by extracting functions. 
Yet the code is quite sensitive and complicated while github diff doesn't show changes well.
It would help if you split this PR into 2-3 smaller commits to see clearly which part of the code goes to which function.

**Comment by pgarg66** [comment]
> Generally, it is a good idea to simplify original code by extracting functions. Yet the code is quite sensitive and complicated while github diff doesn't show changes well. It would help if you split this PR into 2-3 smaller commits to see clearly which part of the code goes to which function.

@KirillLykov, thanks for reviewing this PR. I have tried to functionally split the PR into three commits. Hopefully it makes it easier to view the code diff. The first two commits contain the most relevant diffs.

File: streamer/src/nonblocking/quic.rs [hunk 1]
```diff
@@ -205,20 +237,25 @@ async fn setup_connection(
...
-                connection,
-                uni_streams,
-                ..
-            } = new_connection;
 
-            let remote_addr = connection.remote_address();
+            let remote_addr = new_connection.connection.remote_address();
             let mut remote_pubkey = None;
 
             let table_and_stake = {
```
Comment (KirillLykov) [inline streamer/src/nonblocking/quic.rs]: Looks like `table_and_stake` is not used and it will be removed in follow up commits

**Review by KirillLykov — COMMENTED** [review]
I looked through it and didn't found any suspicious changes. Yet I would ask contributors to this part of code to check it as well

**Review by sakridge — APPROVED** [review]
lgtm

**Pull request merged**